libraries/ms-common/inc/soft_timer.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <stdint.h>
+
+#include "status.h"
+
+#define SOFT_TIMER_NUM 10
+
+typedef void (*soft_timer_callback)(uint8_t timer_id, void* context);
+
+// Initializes a set of software timers. Clock speed should be in MHz. Subsequent calls will do
+// nothing. The clock speed is that of the external PLL crystal.
+void soft_timer_init(uint8_t clock_speed);
```

Comment:
Yeah, I was just thinking there are multiple modules interested in the peripheral clockspeed and it might be nice for debugging